src/routes/backingImage/BackupBackingImage.js

```diff
@@ -68,13 +87,23 @@ function BackupBackingImage({ backingImage, loading, dispatch, location, classNa
       { value: 'url', name: 'URL' },
     ],
     onSearch(filter) {
-      dispatch({
-        type: 'backingImage/setSearchFilter',
-        payload: {
-          bbiSearchField: filter.field,
-          bbiSearchValue: filter.value,
-        },
-      })
+      const { field, value } = filter
+
+      if (persistFilterInURL) {
+        const trimmedValue = value?.trim()
+        const params = (field && trimmedValue)
+          ? { ...queryString.parse(search), field, value: trimmedValue }
+          : {}
+
+        dispatch(routerRedux.push({
+          pathname,
+          hash,
+          search: queryString.stringify(params)
+        }))
+      } else {
+        setBbiSearchField(filter.field)
+        setBbiSearchValue(filter.value)
+      }
```

Comment:
If `persistFilterInURL` is `true`, the search filter will be attached to the URL (Backup page), otherwise, it will be in table search (Backing Image page).